From 9dfef757f6e0da7b1150cfe5a14c676368649692 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 26 Mar 2018 15:24:03 +0800
Subject: [PATCH 3591/5242] MLK-17931-3 gpu: imx: dpu: common: Add
 dpu_has_prefetch_fixup() helper support

commit  051ca3a81cf2ae60ef3e4ac20f6df04fc5f73b30 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds dpu_has_prefetch_fixup() helper support.
Users may use it to tell if a DPU has fixups for prefetch
engines in silicon or not.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-common.c |    6 ++++++
 include/video/dpu.h              |    2 ++
 2 files changed, 8 insertions(+)

diff --git a/drivers/gpu/imx/dpu/dpu-common.c b/drivers/gpu/imx/dpu/dpu-common.c
index 3d66ddf..b580dfa 100644
--- a/drivers/gpu/imx/dpu/dpu-common.c
+++ b/drivers/gpu/imx/dpu/dpu-common.c
@@ -619,6 +619,12 @@ static inline void dpu_cm_write(struct dpu_soc *dpu, u32 value,
 };
 MODULE_DEVICE_TABLE(of, dpu_dt_ids);
 
+bool dpu_has_prefetch_fixup(struct dpu_soc *dpu)
+{
+	return dpu->devtype->has_prefetch_fixup;
+}
+EXPORT_SYMBOL_GPL(dpu_has_prefetch_fixup);
+
 bool dpu_vproc_has_fetcheco_cap(u32 cap_mask)
 {
 	return !!(cap_mask & DPU_VPROC_CAP_FETCHECO);
diff --git a/include/video/dpu.h b/include/video/dpu.h
index 6038e07..99b2b6a 100644
--- a/include/video/dpu.h
+++ b/include/video/dpu.h
@@ -793,6 +793,8 @@ void layerblend_pixengcfg_clken(struct dpu_layerblend *lb,
 struct dpu_hscaler *fetchdecode_get_hscaler(struct dpu_fetchdecode *fd);
 struct dpu_vscaler *fetchdecode_get_vscaler(struct dpu_fetchdecode *fd);
 
+bool dpu_has_prefetch_fixup(struct dpu_soc *dpu);
+
 bool dpu_vproc_has_fetcheco_cap(u32 cap_mask);
 bool dpu_vproc_has_hscale_cap(u32 cap_mask);
 bool dpu_vproc_has_vscale_cap(u32 cap_mask);
-- 
1.7.9.5

